samples/hello_cloudevents/google_events.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * This file contains proposed modifications to the @google/events package. The cloud
+ * function samples in ./src import this file in lieu of the @google/events npm package
+ * so they use the amended API.
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import {LogEntryData as CloudAuditV1LogEntryData} from '@google/events/cloud/audit/v1/LogEntryData';
+import {BuildEventData as CloudCloudbuildV1BuildEventData} from '@google/events/cloud/cloudbuild/v1/BuildEventData';
+import {DocumentEventData as CloudFirestoreV1DocumentEventData} from '@google/events/cloud/firestore/v1/DocumentEventData';
+import {MessagePublishedData as CloudPubsubV1MessagePublishedData} from '@google/events/cloud/pubsub/v1/MessagePublishedData';
+import {SchedulerJobData as CloudSchedulerV1SchedulerJobData} from '@google/events/cloud/scheduler/v1/SchedulerJobData';
+import {StorageObjectData as CloudStorageV1StorageObjectData} from '@google/events/cloud/storage/v1/StorageObjectData';
+import {AnalyticsLogData as FirebaseAnalyticsV1AnalyticsLogData} from '@google/events/firebase/analytics/v1/AnalyticsLogData';
+import {AuthEventData as FirebaseAuthV1AuthEventData} from '@google/events/firebase/auth/v1/AuthEventData';
+import {ReferenceEventData as FirebaseDatabaseV1ReferenceEventData} from '@google/events/firebase/database/v1/ReferenceEventData';
+import {RemoteConfigEventData as FirebaseRemoteconfigV1RemoteConfigEventData} from '@google/events/firebase/remoteconfig/v1/RemoteConfigEventData';
+import {TestMatrixEventData as FirebaseTestlabV1TestMatrixEventData} from '@google/events/firebase/testlab/v1/TestMatrixEventData';
+
+/**
+ * Modification 1: The @google/events package uses the CloudEvent interface provided by
+ * the CloudEvents SDK. The mechanism by which it obtains this interface is unimportant
+ * but it must match the CloudEvent interface used by the Functions Framework.
+ */
+import {CloudEventV1 as CloudEvent} from 'cloudevents';
+
+/**
+ * Modification 2: All components of the public API (i.e. the data schema interfaces) should
+ * be exported from the package root (this is a best practice when publishing NPM packages).
+ */
+export {
+  CloudAuditV1LogEntryData,
+  CloudCloudbuildV1BuildEventData,
+  CloudFirestoreV1DocumentEventData,
+  CloudPubsubV1MessagePublishedData,
+  CloudSchedulerV1SchedulerJobData,
+  CloudStorageV1StorageObjectData,
+  FirebaseAnalyticsV1AnalyticsLogData,
+  FirebaseAuthV1AuthEventData,
+  FirebaseDatabaseV1ReferenceEventData,
+  FirebaseRemoteconfigV1RemoteConfigEventData,
+  FirebaseTestlabV1TestMatrixEventData,
+};
+
+/**
+ * Modification 3: Declare a mapped type that defines which CloudEvent types correspond
+ * to which data payload schemas. This is an internal implementation detail of the
+ * @google/events library, it is not exposed in the public API.
```

Comment:
Is this type statement proposed for addition to https://github.com/googleapis/google-cloudevents-nodejs? Initially reacting to the comment, but more generally wondering how many of these type behaviors should we support outside the functions framework?

Reply:
Yes the proposal is to that all the changes in this file (including this type) would be implemented in the googleapis/google-cloudevents-nodejs library. GCF customers working with cloudevents could choose to take a dependency on that library to get better type checking / autocompletion of their CloudEvent functions.